### PR TITLE
feat(tasks): using TypeScript

### PR DESCRIPTION
**Pull request by emidev98**
Change lib and tasks files from JavaScript to TypeScript to improve the syntax checking

### CODE REVIEW

File: src/commands/task/run.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default class Run extends Command {
     const fromCwd = (p: string) => path.join(process.cwd(), p);
 
     runScript(
-      fromCwd(`tasks/${args.task}.js`),
+      fromCwd(`tasks/${args.task}.ts`),
```
Comment (octalmage) [inline src/commands/task/run.ts:46]: Would it be possible to support both TypeScript or JavaScript? Don’t want to break existing projects.

Comment (emidev98) [reply src/commands/task/run.ts:46]: Possible approaches: 

- Framework generator, where you can create your initial setup with JS or TS
- Accept both TS and JS with a deprecation flag for JS (delete js support in x few versions)
- Accept both TS and JS with a fallback on JS (by default create the files .ts)

Which one you like? Or how would you like to approach accepting both js and ts?

Comment (octalmage) [reply src/commands/task/run.ts:46]: I like the last option, with the task name provided we can check to see which file extension exists in the repo.

**Review by octalmage — APPROVED** [review]
Looks good man! Thanks!

**Pull request merged**